.travis.yml

```diff
@@ -1,25 +1,68 @@
+dist: xenial
+
+os: linux
+
 language: ruby
+
+cache:
+  bundler: true
+
 before_install:
   - gem uninstall -v '>= 2' -i $(rvm gemdir)@global -ax bundler || true
   - gem install bundler -v '< 2'
+
 rvm:
   - ruby-head
   - 3.0
   - 2.7
   - 2.6
+
 gemfile:
-  - gemfiles/rails_5.gemfile
-  - gemfiles/rails_5.1.gemfile
-  - gemfiles/rails_5.2.gemfile
-  - gemfiles/rails_6.0.gemfile
+  - gemfiles/rails_edge.gemfile
   - gemfiles/rails_6.1.gemfile
-matrix:
+  - gemfiles/rails_6.0.gemfile
+  - gemfiles/rails_5.2.gemfile
+  - gemfiles/rails_5.1.gemfile
+  - gemfiles/rails_5.0.gemfile
+
+jobs:
   allow_failures:
     - rvm: ruby-head
+    - gemfile: gemfiles/rails_edge.gemfile
+
+  exclude:
+    - rvm: ruby-head
+      gemfile: gemfiles/rails_5.2.gemfile
+    - rvm: ruby-head
+      gemfile: gemfiles/rails_5.1.gemfile
+    - rvm: ruby-head
+      gemfile: gemfiles/rails_5.0.gemfile
+
+    - rvm: 3.0
+      gemfile: gemfiles/rails_5.2.gemfile
     - rvm: 3.0
       gemfile: gemfiles/rails_5.1.gemfile
     - rvm: 3.0
+      gemfile: gemfiles/rails_5.0.gemfile
+
+    - rvm: 2.7
```

Comment:
Rails 5.1 and 5.2 seem to work with Ruby 2.6 and 2.7, they only fail on Ruby 3 with Sprockets 3. I'd like to revert changes in this section of the file

Reply:
Let me know if this works for you.

I've excluded 3.0 stable tests against Rails 5.x, since it is not officially supported and it can easily save 3 builds. It will preserve testing against ruby-edge, but in that case failures are allowed.

I think that tests should not be allowed to fail against the latest stable version of Ruby